x/concentrated-liquidity/lp.go

```diff
@@ -133,6 +136,16 @@ func (k Keeper) withdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 		return sdk.Int{}, sdk.Int{}, err
 	}
 
+	// If underlying lock exists, validate unlocked conditions are met before withdrawing liquidity.
+	// If unlocked conditions are met, remove the link between the position and the underlying lock.
+	underlyingLockId, _ := k.GetPositionIdToLock(ctx, positionId)
+	if underlyingLockId != 0 {
```

Comment:
If I'm not mistaken, even though LPs aren't able to withdraw until lock maturity, they still are forced to forfeit incentives if they claim early.

Thoughts on also adding a check to `collectIncentives` that does not forfeit rewards if the LP is locked in? Only case I can think of that we'd need to keep in mind is for locks that are already unbonding (and therefore might have <2 weeks of lockup)

Reply:
Sorry, I am not understanding this comment. If someone tries to withdraw a position that is still locked, they will run into this logic gate, fail, and nothing will happen (they will not forfeit their incentives)